### ISSUE TITLE
Custom plugins written in TS don't work since v1.12

### ISSUE THREAD

**Issue by vhenzl** [dotansimha/graphql-code-generator#3849]
Previously it was possible to write and use custom plugins in TypeScript directly without transpiling them to JS by using `ts-node` but id doesn't work since v1.12:

```
 Found 1 error

  × src/types.ts
    Unable to load template plugin matching 'src/plugins/my-test-plugin.ts'.
    Reason:
      Cannot use import statement outside a module
    Error: Unable to load template plugin matching src/plugins/my-test-plugin.ts
        at getPluginByName (C:\Data\Projects\temp\gqlgen-ts-plugin-repro\node_modules\@graphql-codegen\cli\bin.js:230:23)
    Error: Unable to load template plugin matching src/plugins/my-test-plugin.ts
        at getPluginByName (C:\Data\Projects\temp\gqlgen-ts-plugin-repro\node_modules\@graphql-codegen\cli\bin.js:230:23)
```

**To Reproduce**

A simple plugin (in `src/plugins/my-test-plugin.ts`):

```ts
import { GraphQLSchema } from 'graphql';
import { PluginFunction, Types } from '@graphql-codegen/plugin-helpers';

export const plugin: PluginFunction<any> = async (
  schema: GraphQLSchema,
  documents: Types.DocumentFile[],
  config: any
): Promise<Types.PluginOutput> => {
  return {
    content: '',
    prepend: ["// " + new Date().toISOString() + "\n"],
  };
};

export default { plugin };
```

`codegen.yml` config file:

```yml
require:
  - ts-node/register
schema: schema.graphql
documents: './src/**/*.{graphql,js,jsx,ts,tsx}'
generates:
  src/types.ts:
    - src/plugins/my-test-plugin.ts
    - typescript
    - typescript-operations
```

Working in v1.11.2: https://codesandbox.io/s/graphql-codegen-ts-plugin-v1112-ok-n4yg5
Broken in v1.12.1: https://codesandbox.io/s/graphql-codegen-ts-plugin-v1121-error-c0xrn

The error in the sandbox is different than the above I get locally but the outcome is the same – it can't load a TS file.

**Expected behavior**
To be able to use custom plugins written in TS as it was in v1.11.2 and before.

**Environment:**

- OS: Win 10
- `@graphql-codegen/...`: v1.12.1+
- NodeJS: v12

**Additional context**

Not sure if it's relevant but I found this change done in #3328 which seems to be removing support for `ts-node` and all other `require` extensions:

https://github.com/dotansimha/graphql-code-generator/pull/3328/files#diff-4c1eede51cbf2b11c42b7dd505820cd5L68-L73

**Comment by dotansimha** [comment]
We moved the logic of loading require extensions to the `CodeFileLoader`, which effects only while loading schemas/documents from `.ts` files.  This was a mistake :) 

I tried to fix it in: https://github.com/dotansimha/graphql-code-generator/pull/3868

I tested the alpha version of that PR (`1.13.3-alpha-1fc42778.28+1fc42778`) on your repo and it seems to work :)

**Comment by vhenzl** [comment]
I have tried 1.13.3 alpha version on my project and it works. Thanks!

**Comment by dotansimha** [comment]
Fixed in 1.13.3

**Comment by kusmierz** [comment]
Hm, guys, I guess it is broken again in 1.14.0...
